sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/SymmetricKeyCryptographyClient.java

```diff
@@ -72,24 +73,21 @@
 
         byte[] iv = encryptOptions.getIv();
         byte[] additionalAuthenticatedData = encryptOptions.getAdditionalAuthenticatedData();
-        byte[] authenticationTag = generateRandomByteArray(GCM_NONCE_SIZE);
 
         if (iv == null) {
-            if (algorithm == EncryptionAlgorithm.A128GCM || algorithm == EncryptionAlgorithm.A192GCM
-                || algorithm == EncryptionAlgorithm.A256GCM) {
-
+            if (isGcm(algorithm)) {
                 iv = generateRandomByteArray(GCM_NONCE_SIZE);
```

Comment:
I don't think a random nonce (even generated by a cryptographic RNG) is considered secure with AESGCM. Because of this I believe we decided not to do any local AESGCM encryption and always defer to the service. @heaths could you confirm?

Reply:
You are right, I just talked to Heath and he confirmed this, thanks!

Reply:
I will remove the local GCM implementation in a separate PR.

Reply:
yea it was decided in the arch board meeting that we won't be doing local GCM encryption due to random IVs not being secure.